WordPress/gutenberg
# Fix/50850 Global Style applied correctly to the table block

Closes #50850

## What?

This PR fixes the issue between the global styles and the block editor styles for the table block. Also refactors the `block.json` to use [the new Selectors API](https://make.wordpress.org/core/2023/07/17/introducing-the-block-selectors-api/).

## Why?

The global styles were only applied to the table tag but a few styles like spacing should be applied to the td tags.

## How?

This PR fixes the issue by updating the global styles CSS and the block editor-level styles and also by using the new Selectors API.

## Testing Instructions

1. Go to the site editor.
2. Add a table block with some contents
3. Go to Global Styles and change the table block border globally (color and width.)
4. See the global style changes are applied correctly.
5. Also confirm the global styles are overwritten at block settings level.

## Screenshots or screencast 

![image](https://github.com/WordPress/gutenberg/assets/25176325/33289ae5-fa7e-4576-b77b-a5b8e328057e)
